semantic-release/semantic-release
# ES6 migration, repo harminization

- ES 6 + AVA + test refactoring:
    - Use async/await instead of callbacks
    - Use [execa](https://github.com/sindresorhus/execa) to run command lines
    - Use AVA for tests
    - Add several assertions in existing unit tests
    - Add documentation (comments) in the tests
    - Run tests with a real git repo instead of mocking `child_process` and add test helpers to create repos, commits and checkout
    - Simplify test directory structure
    - Simplify code readability (mostly with async/await)
    - Switch from `standard` to `eslint` + `eslint-config-standard` + [prettier](https://github.com/prettier/prettier)
- Add a License file + badge
- Remove npm and yarn lockfile
- Use a more generic `.gitignore` that would work on Windows, Linux, Mac OS
- Add badges

The test coverage appear to be worse than before because `src/index.js` was not included in the coverage and it is now. The coverage is actually a bit better now.
I'll add more tests in a following PR.

Overall this PR doesn't change functionality, and does not fix bugs. It makes code and unit/integration tests more readable and easier to understand, hopefully making the project more accessible to contributors.

I found several improvement and bugs to fix will rewriting the tests. That will come in following PRs.

Fix #391, Fix #378, Fix #361
